website/www/site/content/en/documentation/transforms/python/elementwise/mltransform.md

```diff
@@ -0,0 +1,111 @@
+with beam.Pipeline() as p:
+  (data | MLTransform(artifact_location).with_transform(scale_to_z_score_transform))
+```
+
+In this example, `MLTransform` receives `artifact_location`. This location is used to store artifacts generated by the `MLTransform`. We will talk about the `artifacts` later. The data processing transform can be passed using the `with_transform` method of `MLTransform` or transforms can be passed as list to the MLTransform.
+
+```
+MLTransform(transforms=transforms, artifact_location=artifact_location)
+```
+
+All the transforms passed to `MLTransform` are applied sequentially on the dataset. `MLTransform` expects a keyed data (Link to the supported inputs).
+
+
+## Example 1
+
+In the example we will create a pipeline that uses MLTransform to scale the data between 0 and 1.
+
+{{< highlight language="py" file="sdks/python/apache_beam/examples/snippets/transforms/elementwise/mltransform.py"
+  class="notebook-skip" >}}
+{{< code_sample "sdks/python/apache_beam/examples/snippets/transforms/elementwise/mltransform.py" mltransform_scale_to_0_1 >}}
+{{</ highlight >}}
+
+{{< paragraph class="notebook-skip" >}}
+Output:
+{{< /paragraph >}}
+{{< highlight class="notebook-skip" >}}
+{{< code_sample "sdks/python/apache_beam/examples/snippets/transforms/elementwise/mltransform_test.py" mltransform_scale_to_0_1 >}}
+{{< /highlight >}}
+
+
+The example takes a list of ints and converts them into the range of 0 to 1 using the transform `ScaleTo01`.
+
+## Example 2
+
+In the example, we will create a pipeline that use MLTransform to compute vocabulary on the entire dataset and assign indices to each unique vocab.
+
+{{< highlight language="py" file="sdks/python/apache_beam/examples/snippets/transforms/elementwise/mltransform.py"
+  class="notebook-skip" >}}
+{{< code_sample "sdks/python/apache_beam/examples/snippets/transforms/elementwise/mltransform.py" mltransform_compute_and_apply_vocabulary >}}
+{{</ highlight >}}
+
+{{< paragraph class="notebook-skip" >}}
+Output:
+{{< /paragraph >}}
+{{< highlight class="notebook-skip" >}}
+{{< code_sample "sdks/python/apache_beam/examples/snippets/transforms/elementwise/mltransform_test.py" mltransform_compute_and_apply_vocabulary >}}
+{{< /highlight >}}
+
+
+This example takes a list of strings, computes vocab over the entire dataset and applies a unique index to each vocab.
+
+
+The above two examples requires a full pass over the dataset to transform the dataset. For `ComputeAndApplyVocabulary`, all the unqiue words in the dataset needs to be known before transforming the data. For `ScaleTo01`, the minimum and maximum of the dataset needs to be known before transforming the dataset. This is acheived by `MLTransform`.
+
+
+In this example, we will create a pipeline that use MLTransform to compute vocabulary on the entire dataset and assign indices to each unique vocab. This pipeline takes a single element as input instead of list of elements.
```

Comment:
```suggestion
This example creates a pipeline that uses `MLTransform` to compute vocabulary on the entire dataset and assign indices to each unique vocabulary item. This pipeline takes a single element as input instead of a list of elements.
```